packages/salesforce-adapter/test/filters/audit_information.test.ts

```diff
@@ -0,0 +1,150 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { CORE_ANNOTATIONS, ElemID, Element, ObjectType, PrimitiveType, PrimitiveTypes, ReferenceExpression, InstanceElement } from '@salto-io/adapter-api'
+import mockClient from '../client'
+import Connection from '../../src/client/jsforce'
+import SalesforceClient from '../../src/client/client'
+import * as utils from '../../src/filters/utils'
+import { Filter, FilterResult } from '../../src/filter'
+import auditInformation, { WARNING_MESSAGE } from '../../src/filters/audit_information'
+import { defaultFilterContext, MockInterface } from '../utils'
+import { buildFetchProfile } from '../../src/fetch_profile/fetch_profile'
+import { API_NAME, CUSTOM_FIELD, CUSTOM_OBJECT } from '../../src/constants'
+import * as transformer from '../../src/transformers/transformer'
+
+
+describe('audit information test', () => {
+  let filter: Filter
+  let client: SalesforceClient
+  let connection: MockInterface<Connection>
+  let customObject: ObjectType
+  const objectProperties = { fullName: 'Custom__c',
+    createdByName: 'created_name',
+    createdDate: 'created_date',
+    lastModifiedByName: 'changed_name',
+    lastModifiedDate: 'changed_date' }
+  const fieldProperties = { fullName: 'Custom__c.StringField__c',
+    createdByName: 'created_name_field',
+    createdDate: 'created_date_field',
+    lastModifiedByName: 'changed_name_field',
+    lastModifiedDate: 'changed_date_field' }
+  const nonExistentFieldProperties = { fullName: 'Custom__c.noSuchField',
+    createdByName: 'test',
+    createdDate: 'test',
+    lastModifiedByName: 'test',
+    lastModifiedDate: 'test' }
+  // In order to test a field that was described in the server and not found in our elements.
+  const primID = new ElemID('test', 'prim')
+  const primNum = new PrimitiveType({
+    elemID: primID,
+    primitive: PrimitiveTypes.STRING,
+    annotationRefsOrTypes: {},
+    annotations: {},
+  })
+  const checkElementAnnotations = (object: Element, properties: Record<string, string>): void => {
+    expect(object.annotations[CORE_ANNOTATIONS.CREATED_BY]).toEqual(properties.createdByName)
+    expect(object.annotations[CORE_ANNOTATIONS.CREATED_AT]).toEqual(properties.createdDate)
+    expect(object.annotations[CORE_ANNOTATIONS.CHANGED_BY]).toEqual(properties.lastModifiedByName)
+    expect(object.annotations[CORE_ANNOTATIONS.CHANGED_AT]).toEqual(properties.lastModifiedDate)
+  }
+  const objectWithoutInformation = new ObjectType({
+    elemID: new ElemID('salesforce', 'otherName'),
+    annotations: { metadataType: 'CustomObject', [API_NAME]: 'otherName' },
+    fields: {
+      StringField__c: { refType: new ReferenceExpression(primNum.elemID, primNum) },
+    },
+  })
+  // In order to test an object without audit information in server.
+
+  beforeEach(() => {
+    ({ connection, client } = mockClient())
+    connection.metadata.list = jest.fn()
```

Comment:
connection is already a mock interface IIRC, you should not replace its functions (that loses type information by using the untyped `jest.fn()`)
```suggestion
    connection.metadata.list.mockImplementation(...)
```

Reply:
its already a mock but the current mock expect the return value to be a full fileProperties value
so this works fine.

Reply:
that is the whole point of having typed mocks - things are not fine when you return a value that can never happen as part of your test because it means you are testing your code in scenarios that will never happen...

Reply:
fixed.